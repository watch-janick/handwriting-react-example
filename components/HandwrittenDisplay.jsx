/**
 * Component used to display the rendered PNG
 */
const HandwrittenDisplay = ({ image }) => (
  <div id="handwritten-image">
    <img src={image} />
  </div>
)

export default HandwrittenDisplay