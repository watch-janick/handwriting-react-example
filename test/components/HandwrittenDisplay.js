import HandwrittenDisplay from '../../components/HandwrittenDisplay'

describe('HandwrittenDisplay', () => {
  it('should render correctly with image URL', () => {
    const attrs = {
      image: 'https://images.unsplash.com/photo-1530518309799-9db5fc42276b?ixlib=rb-0.3.5…EyMDd9&s=b784735…&auto=format&fit=crop&w=2275&q=80',
    }
    const component = shallow(<HandwrittenDisplay {...attrs} />)

    expect(component).to.matchSnapshot()
  })
})