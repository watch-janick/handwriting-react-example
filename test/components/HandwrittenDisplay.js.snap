// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandwrittenDisplay should render correctly with image URL 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HandwrittenDisplay
    image="https://images.unsplash.com/photo-1530518309799-9db5fc42276b?ixlib=rb-0.3.5…EyMDd9&s=b784735…&auto=format&fit=crop&w=2275&q=80"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        src="https://images.unsplash.com/photo-1530518309799-9db5fc42276b?ixlib=rb-0.3.5…EyMDd9&s=b784735…&auto=format&fit=crop&w=2275&q=80"
      />,
      "id": "handwritten-image",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "src": "https://images.unsplash.com/photo-1530518309799-9db5fc42276b?ixlib=rb-0.3.5…EyMDd9&s=b784735…&auto=format&fit=crop&w=2275&q=80",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          src="https://images.unsplash.com/photo-1530518309799-9db5fc42276b?ixlib=rb-0.3.5…EyMDd9&s=b784735…&auto=format&fit=crop&w=2275&q=80"
        />,
        "id": "handwritten-image",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "https://images.unsplash.com/photo-1530518309799-9db5fc42276b?ixlib=rb-0.3.5…EyMDd9&s=b784735…&auto=format&fit=crop&w=2275&q=80",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
